analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RuleChecks/InitializationVectorShouldBeRandomBase.cs

```diff
@@ -18,10 +18,108 @@
  * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
 
+using System.Security.Cryptography;
+using SonarAnalyzer.SymbolicExecution.Constraints;
+
 namespace SonarAnalyzer.SymbolicExecution.Roslyn.RuleChecks;
 
 public abstract class InitializationVectorShouldBeRandomBase : SymbolicRuleCheck
 {
     protected const string DiagnosticId = "S3329";
     protected const string MessageFormat = "Use a dynamically-generated, random IV.";
+
+    protected override ProgramState PreProcessSimple(SymbolicContext context)
+    {
+        var state = context.State;
+        if (context.Operation.Instance.AsArrayCreation() is { } arrayCreation
+            && arrayCreation.Type.Is(KnownType.System_Byte_Array)
+            && arrayCreation.DimensionSizes.Length == 1)
+        {
+            return state.SetOperationConstraint(arrayCreation, ByteCollectionConstraint.CryptographicallyWeak);
+        }
+        else if (context.Operation.Instance.AsAssignment() is { } assignment)
+        {
+            return ProcessAssignmentToIVProperty(assignment, state)
+                   ?? state;
+        }
+        else if (context.Operation.Instance.AsInvocation() is { } invocation)
+        {
+            return ProcessStrongRandomGeneratorMethodInvocation(invocation, state)
+                    ?? ProcessCreateEncryptorMethodInvocation(invocation, state, context)
+                    ?? state;
+        }
+        else
+        {
+            return state;
+        }
+    }
+
+    private static ProgramState ProcessAssignmentToIVProperty(IAssignmentOperationWrapper assignment, ProgramState state)
+    {
+        return assignment.Target?.AsPropertyReference() is { } property
+                && GetIVPropertyInstance(property) is { } instanceSymbol
+                && (assignment.Value.AsArrayCreation() is { } || state[assignment.Value].HasConstraint(ByteCollectionConstraint.CryptographicallyWeak))
+                    ? state.SetSymbolConstraint(instanceSymbol, ByteCollectionConstraint.CryptographicallyWeak)
+                    : null;
+
+        static ISymbol GetIVPropertyInstance(IPropertyReferenceOperationWrapper property) =>
+            property.Property.Name.Equals("IV")
+            && property.Instance.TrackedSymbol() is { } instanceSymbol
+            && instanceSymbol.GetSymbolType().DerivesFrom(KnownType.System_Security_Cryptography_SymmetricAlgorithm)
+               ? instanceSymbol
+               : null;
+    }
```

Comment:
A few things:
- Change parameter order to be sorted by creation time (earliest first).
- Do not use a helper method. The code is simple enough without it.
- Use `ResolveCaputreAndUnwrapConversion` to fix FN.
- Remove unnecessary check for array creation (it will have a constraint anyway).

```suggestion
    private static ProgramState ProcessAssignmentToIVProperty(ProgramState state, IAssignmentOperationWrapper assignment) =>
        assignment.Target?.AsPropertyReference() is { } property
            && property.Property.Name.Equals("IV")
            && state.ResolveCaptureAndUnwrapConversion(property.Instance).TrackedSymbol() is { } instanceSymbol
            && instanceSymbol.GetSymbolType().DerivesFrom(KnownType.System_Security_Cryptography_SymmetricAlgorithm)
            && state[assignment.Value].HasConstraint(ByteCollectionConstraint.CryptographicallyWeak)
                ? state.SetSymbolConstraint(instanceSymbol, ByteCollectionConstraint.CryptographicallyWeak)
                : null;
```